### PR TITLE
PO conversion

### PR DESCRIPTION
**Pull request by fangfufu**
We are moving from YAML format to PO format in order to future proof the project, as per previous discussion. I have re-written the README and explain what's happening, and I generate the en-US po files for the original 6 web pages. 

### Open Questions / Tasks
* [x] Decision regarding the right way to handle fully formatted text blocks
  * Do we always exclude Markdown from the translation files if a text block is formatted as a whole (e.g. an entire paragraph is bold)? **Yes**
  * Do we remove links from the translation files if they are not inline, but affect an entire text block? **Yes**
* [x] Decision regarding which meta data we want to have in the files and how they should be formatted
* [x] Final review done regarding encoding issues (e.g. `?` instead of apostrophes, quotes and dashes)

### CODE REVIEW

File: Localization/en-US/faq.po [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Source-Version: 1.0
+# Source-Last-Modified: 20200705
+# Target-Language:
+# Target-Version:
+# Target-Last_Modified:
+
+msgid "FAQ"
+msgstr ""
+
+msgid "**You?ll find a large library of frequently asked questions about all things connected to Folding@home.**"
```
Comment (Tar-Minyatur) [inline Localization/en-US/faq.po]:
```suggestion
msgid "You'll find a large library of frequently asked questions about all things connected to Folding@home."
```

Looks like there is a little problem with special characters. That should probably be `You'll` ;)

Also...as this entire statement is bold, maybe we don't extract it with Markdown, but assume wherever the string goes it will be made bold by the layout?

File: Localization/en-US/home.po [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+msgid "I AM"
+msgstr ""
+
+msgid "One"
+msgstr ""
+
+msgid "IN A MILLION"
+msgstr ""
+
+msgid "Regardless if you are already folding or haven?t heard a word about it before, we need your help to reach our goal ? which is 1 million folders."
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Regardless if you are already folding or haven't heard a word about it before, we need your help to reach our goal – which is 1 million folders."
```

File: Localization/en-US/home.po [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+msgstr ""
+
+msgid "Start folding now"
+msgstr ""
+
+msgid "[Hide this and watch the video.](#)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Hide this and watch the video."
```

I guess this Link contains some JavaScript, which we cannot cover this way, so we'll have to exclude the link from the Markdown.

File: Localization/en-US/home.po [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+msgid "Linux"
+msgstr ""
+
+msgid "Windows"
+msgstr ""
+
+msgid "Mac"
+msgstr ""
+
+msgid "[Start folding](https://foldingathome.org/start-folding/)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Start folding"
```

Same here...I think there is no point to do the link extraction if the link sits on its own and not as an inline formatting, right?

File: Localization/en-US/home.po [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+msgstr ""
+
+msgid "Latest Posts"
+msgstr ""
+
+msgid "[The SARS-CoV-2 nucleocapsid protein is dynamic, disordered, and phase separates with RNA.](https://foldingathome.org/2020/06/26/the-sars-cov-2-nucleocapsid-protein-is-dynamic-disordered-and-phase-separates-with-rna/)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]: Should we really include these? They're probably auto-generated, I would guess.

File: Localization/en-US/home.po [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+msgid "[The Covid Moonshot](https://foldingathome.org/2020/05/28/the-covid-moonshot/)"
+msgstr ""
+
+msgid "JOIN THE COMMUNITY"
+msgstr ""
+
+msgid "The Folding@Home community is spread over countless forums, websites and social media."
+msgstr ""
+
+msgid "Here?s a list of our official channels."
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Here's a list of our official channels."
```

File: Localization/en-US/home.po [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+msgid "[Twitter](https://twitter.com/foldingathome)"
+msgstr ""
+
+msgid "1 in a million"
+msgstr ""
+
+msgid "WHY WE NEED YOU"
+msgstr ""
+
+msgid "Folding@home is a project focused on disease research. The problems we?re solving require so many computer calcul�ations ? and we need your help to find the cures!"
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Folding@home is a project focused on disease research. The problems we're solving require so many computer calculations – and we need your help to find the cures!"
```

File: Localization/en-US/home.po [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+msgid "Fight diseases Together with us."
+msgstr ""
+
+msgid "The Folding@home software allows you to share your unused computer power with us ? so that we can research even more potential cures."
+msgstr ""
+
+msgid "FOLDING?"
+msgstr ""
+
+msgid "Folding refers to the way human protein folds in the cells that make up your body. We rely on the proteins to keep us healthy and they assemble themselves by folding. But when they misfold, there can be serious consequences to a person?s health."
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Folding refers to the way human protein folds in the cells that make up your body. We rely on the proteins to keep us healthy and they assemble themselves by folding. But when they misfold, there can be serious consequences to a person's health."
```

File: Localization/en-US/home.po [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+msgstr ""
+
+msgid "[Learn more](http://foldingathome.org/diseases/)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Learn more"
```

Again, probably better to only have the string here? Maybe add a comment to explain where this goes?

File: Localization/en-US/start-folding.po [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+# Source-Version: 1.0
+# Source-Last-Modified: 20200705
+# Target-Language:
+# Target-Version:
+# Target-Last_Modified:
+
+msgid "Start folding"
+msgstr ""
+
+msgid "**The Folding@home software runs while you do other things.**"
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "The Folding@home software runs while you do other things."
```

Again with the unnecessary formatting here.

File: Localization/en-US/start-folding.po [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+msgstr ""
+
+msgid "While you keep going with your everyday activities, your computer will be working to help us find cures for diseases like cancer, ALS, Parkinson?s, Huntington?s, Influenza and many others."
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "While you keep going with your everyday activities, your computer will be working to help us find cures for diseases like cancer, ALS, Parkinson's, Huntington's, Influenza and many others."
```

File: Localization/en-US/start-folding.po [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+msgstr ""
+
+msgid "**Please be patient if you experience idle time as we face the COVID-19 pandemic together!** We are working like mad to start simulations of COVID-19 proteins. The enthusiastic response to our effort has been tremendous and has sometimes emptied our queues. This is a great ?problem? to have as there is no end to the valuable science we can do, its just a matter of giving us some time to get more running."
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "**Please be patient if you experience idle time as we face the COVID-19 pandemic together!** We are working like mad to start simulations of COVID-19 proteins. The enthusiastic response to our effort has been tremendous and has sometimes emptied our queues. This is a great “problem“ to have as there is no end to the valuable science we can do, its just a matter of giving us some time to get more running."
```

File: Localization/en-US/start-folding.po [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+msgstr ""
+
+msgid "**Don?t see the correct software for your device? Try our list of [alternative downloads](alternative-downloads).**"
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "Don't see the correct software for your device? Try our list of [alternative downloads](/alternative-downloads)."
```
That link also looked a bit strange.

File: Localization/en-US/start-folding.po [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+msgstr ""
+
+msgid "**Additional platforms**"
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "Additional platforms"
```

File: Localization/en-US/start-folding.po [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+msgstr ""
+
+msgid "[Docker for GPUs ](https://hub.docker.com/r/foldingathome/fah-gpu)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "Docker for GPUs"
```

File: Localization/en-US/start-folding.po [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+msgstr ""
+
+msgid "[vSphere](https://flings.vmware.com/vmware-appliance-for-folding-home)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/start-folding.po]:
```suggestion
msgid "vSphere"
```

File: Localization/en-US/statistics.po [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+msgstr ""
+
+msgid "MAXIMIZE"
+msgstr ""
+
+msgid "your"
+msgstr ""
+
+msgid "**EFFORT**"
+msgstr ""
```
Comment (Tar-Minyatur) [inline Localization/en-US/statistics.po]: This (and the next two) violate the idea of keeping segments together as much as possible. But we would have to ask the designers to change the way they set this up, e.g. by putting the fancy formatting on the `<strong>` tag so that we can do something like this here:
```suggestion
msgid "**MAXIMIZE** your **EFFORT**"
msgstr ""
```

File: Localization/en-US/statistics.po [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+msgid "Set up your team"
+msgstr ""
+
+msgid "If you are interested in setting up a team of your own, you can get started here."
+msgstr ""
+
+msgid "[Start a team](https://apps.foldingathome.org/team/create)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/statistics.po]:
```suggestion
msgid "Start a team"
```

File: Localization/en-US/statistics.po [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+msgstr ""
+
+msgid "[Lost team password?](https://apps.foldingathome.org/team/mailpass)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/statistics.po]:
```suggestion
msgid "Lost team password?"
```

File: Localization/en-US/statistics.po [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+msgstr ""
+
+msgid "[Change your team info](https://apps.foldingathome.org/team/update)"
```
Comment (Tar-Minyatur) [inline Localization/en-US/statistics.po]:
```suggestion
msgid "Change your team info"
```

File: README.md [hunk 21]
```diff
@@ -1,30 +1,134 @@
...
 
-## What do we translate?
-
-We will start translating the [Folding@Home](https://foldingathome.org) website.
-We are starting with the following webpages: 
+## What has changed?
+After the discussion within the translation plugin development team, we have 
+decided to move from the custom YAML format to the more standardised 
+[gettext](https://en.wikipedia.org/wiki/Gettext) Portable Object (PO) format. 
+We have decided to embedding formatting (*not layout*) within the PO files. This
```
Comment (Tar-Minyatur) [inline README.md]:
```suggestion
We have decided to embed formatting (*not layout*) within the PO files. This
```

File: Localization/en-US/home.po (hunk 5 above)
Comment (fangfufu) [reply Localization/en-US/home.po]: I guess we shouldn't.

Comment (fangfufu) [reply Localization/en-US/home.po]: I removed it.

**Review by ghost — COMMENTED** [review]
In README.md, line 45, there seems to be a grammatical error.
"_We are certain Mardown_".

**Comment by fangfufu** [comment]
> In README.md, line 45, there seems to be a grammatical error.
> "_We are certain Mardown_".

I changed it to ``We are using Markdown``.

File: README.md [hunk 22]
```diff
@@ -1,30 +1,134 @@
...
+
+ - strip the menu
+ - escape the double quotes
+ - add the metadata block. 
+ - convert invalid characters, some characters get turned into `?`
+
+We plan to release a proper tool in a later date to generate PO file properly. 
+
+If you do want to add a PO file or modify an existing PO file, please use a **separate** pull request than the normal translation work. This is to maintain
+a clean workflow.
```
Comment (Tar-Minyatur) [inline README.md]:
```suggestion
It's awesome that you're eager to do more translations. For now, however, we don't have an established process yet to add new pages to the translation effort. We have to be careful with how we define the message IDs to ensure that they will later work correctly with our translation system. The translation coordination team will add more files as our processes mature and become more stable.
```

**Comment by fangfufu** [comment]
@Tar-Minyatur , I think I sorted out the metadata problem.

File: Localization/en-US/diseases.po [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+"Source-Version: 1.0\n"
+"Source-Last-Modified: 20200705\n"
+"Target-Language: \n"
+"Target-Version: \n"
+"Target-Last-Modified: \n"
+
+msgid "Diseases"
+msgstr ""
+
+msgid "**The Folding@home project (FAH) is dedicated to understanding protein folding, the diseases that result from protein misfolding and aggregation, and novel computational ways to develop new drugs in general. Here, we briefly describe our goals, what we are doing, and some highlights so far.**"
```
Comment (Tar-Minyatur) [inline Localization/en-US/diseases.po]:
```suggestion
msgid "The Folding@home project (FAH) is dedicated to understanding protein folding, the diseases that result from protein misfolding and aggregation, and novel computational ways to develop new drugs in general. Here, we briefly describe our goals, what we are doing, and some highlights so far."
```

File: Localization/en-US/home.po [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+msgstr ""
+
+msgid "foldingforum.org"
+msgstr ""
+
+msgid "[Facebook](https://www.facebook.com/Foldinghome-136059519794607/)"
+msgstr ""
+
+msgid "[Twitter](https://twitter.com/foldingathome)"
+msgstr ""
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Facebook"
msgstr ""

msgid "Twitter"
msgstr ""
```

We should remove the links here, I think.

File: Localization/en-US/home.po [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+msgid "FOLDING?"
+msgstr ""
+
+msgid "Folding refers to the way human protein folds in the cells that make up your body. We rely on the proteins to keep us healthy and they assemble themselves by folding. But when they misfold, there can be serious consequences to a person's health."
+msgstr ""
+
+msgid "[Learn more](http://foldingathome.org/diseases/)"
+msgstr ""
+
+msgid "**Team**"
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "Team"
```

File: README.md [hunk 26]
```diff
@@ -1,30 +1,124 @@
...
+ - The ``Source-Version`` should follow the format of ``x.y``, where ``x`` and
+ ``y`` are both integers. Increase in ``x`` indicates a major structure
+ change, e.g. adding or removing a paragraph. Increase in ``y`` indicates a
+  minor change, e.g. a spelling mistake. Note that ``x`` and ``y`` can have
+  multiple digits, e.g. ``1.12``. ``y`` should start at 0.
+ - The ``Target-Version`` should also follow the format of ``x.y.z``. ``x`` and
+ ``y`` should follow the ``x`` and ``y`` in ``Original-Version``. ``z`` should
+ start at 0. It should be incremented before pushing the updated PO file to the
+ main repository. ``z`` should be reset to 0, if a new ``Source-Version`` is
+ released.
```
Comment (Tar-Minyatur) [inline README.md:97]: Do we really need this explanation? Can we not just say "please stick to the formats used in the original files"? Seems a bit overwhelming to me.

File: Localization/en-US/home.po [hunk 27]
```diff
@@ -0,0 +1,100 @@
...
+msgid "WHY WE NEED YOU"
+msgstr ""
+
+msgid "Folding@home is a project focused on disease research. The problems we are solving require so many computer calculations and we need your help to find the cures!"
+msgstr ""
+
+msgid "Fight diseases Together with us."
+msgstr ""
+
+msgid "The Folding@home software allows you to share your unused computer power with us ? so that we can research even more potential cures."
```
Comment (Tar-Minyatur) [inline Localization/en-US/home.po]:
```suggestion
msgid "The Folding@home software allows you to share your unused computer power with us – so that we can research even more potential cures."
```

**Comment by Tar-Minyatur** [comment]
@fangfufu Yes, definitely looks like it. I did a test run with POEdit and it works well this way.

I also did my final pass over the msgids to check for more encoding issues. Found one and fixed it.

All that's left now is to decide on those superfluous inline formattings that affect entire text blocks. And there are a couple of other comments left I guess.

File: README.md (hunk 26 above)
Comment (fangfufu) [reply README.md:97]: I think it is better to leave them here, just for future reference. It is useful to know the decision behind the design choice.

We can have a TL;DR version / a shorter version for volunteers.

**Review by Tar-Minyatur — APPROVED** [review]
